src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -170,26 +182,55 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
             }
         }
 
-        internal /*for testing purposes*/ void SilentSaveSolutionIfDirty()
+        internal /*for testing purposes*/ void DownloadBindingParameters(IProgressController controller, CancellationToken token, IProgressStepExecutionEvents notifications)
         {
-            bool saved = VsShellUtils.SaveSolution(this.host, silent: true);
-            Debug.Assert(saved, "Should not be cancellable");
+            notifications.ProgressChanged(Strings.PreparingBindingWorkflowProgessMessage, double.NaN);
+
+            var properties = this.host.SonarQubeService.GetProperties(token);
+
+            if (token.IsCancellationRequested)
+            {
+                AbortWorkflow(controller, token);
+                return;
+            }
+
+            var testProjRegexProperty = properties.FirstOrDefault(x => StringComparer.Ordinal.Equals(x.Key, ServerProperty.TestProjectRegexKey));
+            var testProjRegexPattern = testProjRegexProperty?.Value ?? ServerProperty.TestProjectRegexDefaultValue;
```

Comment:
Except it won't using the old API according to [here](https://jira.sonarsource.com/browse/SONAR-5891) and [here](https://github.com/SonarSource-VisualStudio/sonar-msbuild-runner/blob/c424408125bad657784a11330331e658a6448dd0/SonarQube.TeamBuild.PreProcessor/SonarWebService.cs#L128).